server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/Fixture.scala

```diff
@@ -0,0 +1,94 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.rfc8621.contract
+
+import java.nio.charset.StandardCharsets
+
+import io.restassured.authentication.PreemptiveBasicAuthScheme
+import io.restassured.builder.RequestSpecBuilder
+import io.restassured.config.EncoderConfig.encoderConfig
+import io.restassured.config.RestAssuredConfig.newConfig
+import io.restassured.http.ContentType
+import org.apache.james.GuiceJamesServer
+import org.apache.james.core.{Domain, Username}
+import org.apache.james.jmap.JMAPUrls.JMAP
+import org.apache.james.jmap.draft.JmapGuiceProbe
+import org.apache.james.jmap.http.UserCredential
+
+object Fixture {
+  def baseRequestSpecBuilder(server: GuiceJamesServer) = new RequestSpecBuilder()
+    .setContentType(ContentType.JSON)
+    .setAccept(ContentType.JSON)
+    .setConfig(newConfig.encoderConfig(encoderConfig.defaultContentCharset(StandardCharsets.UTF_8)))
+    .setPort(server.getProbe(classOf[JmapGuiceProbe])
+      .getJmapPort
+      .getValue)
+    .setBasePath(JMAP)
```

Comment:
Is it ok to put that in `Fixture`?

Reply:
IMO yes. This code is duplicated between each tests. And not related much to the logic tested. Abstracting it away sounds reasonable

Reply:
I think it is acceptable for this, but if any kind of `utility methods` are added, we should move it to own class.
Are you agree?

Reply:
Alright, thanks for the explanation